### DISCUSSION TITLE
Not all methods using ExpressionCompiler in RuleExpressionCompiler?

### DISCUSSION THREAD

**Discussion by lustoro** [NRules/NRules#326] [Q&A]
I've started to use NRules and I'm facing performance issues when Compiling rules by using RuleCompiler.Compile(). I've seen that one of the ways to do some performance improvements is using FastExpressionCompiler suggested here https://github.com/NRules/NRules/wiki/Expression-Compiler. But after doing some performance profiling I've seen that it still works slow for CompileLhsFactExpression in RuleExpressionCompiler.cs. And I've seen that CompileLhsFactExpression do not uses ExpressionCompiler for compilation and only that method do not uses ExpressionCompiler, other methods are using ExpressionCompiler
https://github.com/NRules/NRules/blob/f84f59df0f8df32282b980f78686a516169c00b9/src/NRules/NRules/Utilities/RuleExpressionCompiler.cs#L40-L48

to have perfromance improvement is it possible to use ExpressionCompiler for CompileLhsFactExpression  as well ? And have that change in future version ?

**Comment by snikolayev** [comment]
@lustoro thanks for reporting this, really odd how this was missed in the initial implementation. I can confirm that this is a real issue. Will fix shortly

**Comment by snikolayev** [comment]
Fixed by #327

**Reply by lustoro** [reply]
@snikolayev Greate, Thank you!
